engine/src/main/java/org/terasology/engine/rendering/assets/mesh/Mesh.java

```diff
@@ -22,34 +24,26 @@ protected Mesh(ResourceUrn urn, AssetType<?, MeshData> assetType, DisposableReso
 
     public abstract AABBfc getAABB();
 
-    protected AABBf getBound(MeshData data, AABBf dest) {
-        TFloatList vertices = data.getVertices();
-        int vertexCount = vertices.size() / 3;
-        if (vertexCount == 0) {
+    protected AABBf getBound(AABBf dest) {
+        VertexAttributeBinding<Vector3fc, Vector3f> vertices = this.getVertices();
+        if (vertices.getResource().elements() == 0) {
```

Comment:
Should this use the abstraction for `getVertexCount()` here fore consistency?

```suggestion
        if (this.getVertexCount() == 0) {
```